setup.py

```diff
@@ -4,34 +4,148 @@
 #  FreeType high-level python API - copyright 2011 Nicolas P. Rougier
 #  Distributed under the terms of the new BSD license.
 # -----------------------------------------------------------------------------
+from __future__ import absolute_import, print_function
+
+import distutils
+import distutils.dir_util
+import distutils.file_util
+import os
+import subprocess
+import sys
+from io import open
 from os import path
-from codecs import open
-from setuptools import setup, Extension
-
-description = open(path.join(path.abspath(path.dirname(__file__)),
-                             'README.rst'), encoding='utf-8').read()
-
-setup( name        = 'freetype-py',
-       version     = '1.2',
-       description = 'Freetype python bindings',
-       long_description = description,
-       author      = 'Nicolas P. Rougier',
-       author_email= 'Nicolas.Rougier@inria.fr',
-       url         = 'https://github.com/rougier/freetype-py',
-       packages    = ['freetype', 'freetype.ft_enums'],
-       classifiers = [
-          'Development Status :: 5 - Production/Stable',
-          'Environment :: X11 Applications',
-          'Environment :: MacOS X',
-          'Intended Audience :: Developers',
-          'License :: OSI Approved :: GNU General Public License (GPL)',
-          'Operating System :: MacOS',
-          'Operating System :: Microsoft :: Windows',
-          'Operating System :: POSIX',
-          'Operating System :: Unix',
-          'Programming Language :: Python :: 2',
-          'Programming Language :: Python :: 3',
-          'Topic :: Multimedia :: Graphics',
-          ],
-       keywords    = ['freetype', 'font'],
-     )
+
+from setuptools import Extension, find_packages, setup
+from setuptools.command.build_ext import build_ext
+
+cmdclass = {}
+try:
+    from wheel.bdist_wheel import bdist_wheel
+except ImportError:
+    print("warning: wheel package is not installed", file=sys.stderr)
+else:
+
+    class UniversalBdistWheel(bdist_wheel):
+        def get_tag(self):
+            return (
+                'py2.py3',
+                'none',
+            ) + bdist_wheel.get_tag(self)[2:]
+
+    cmdclass['bdist_wheel'] = UniversalBdistWheel
+
+
+class SharedLibrary(Extension):
+    """Object that describes the library (filename) and how to make it.
+
+    Note: The `output_dir` parameter shouldn't include the final lib/ or bin/
+    component, we figure that out on the fly.
```

Comment:
I don't think this class should also determine whether the generated output goes to "lib" or "bin", or the prefix of the library. It's better to leave it agnostic as to the build system used. The library file extension is a different thing, as that is always the same (depending on platforms), and is used in get_ext_filename method to override the default extension module filename.

Reply:
(i like to make this into a reusable setuptools extension which one could use as a build requirement at some point; better not to add too much freetype-py specific logic to it and keep it general)